File: generated-repos/spring-backend/src/main/java/com/helpdesk/service/ActivityLogService.java

package com.helpdesk.service;

import com.helpdesk.dto.ActivityLogDTO;
import com.helpdesk.dto.ActivityLogMapper;
import com.helpdesk.entity.ActivityLog;
import com.helpdesk.repository.ActivityLogRepository;

import com.helpdesk.entity.Ticket;
import com.helpdesk.repository.TicketRepository;





import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import org.springframework.context.ApplicationEventPublisher;

import java.util.List;


@Service
@RequiredArgsConstructor
public class ActivityLogService {

    private final ActivityLogRepository activityLogRepository;

    private final TicketRepository ticketRepository;




    private final ActivityLogMapper activityLogMapper;




    private final ApplicationEventPublisher eventPublisher;




    @Transactional(readOnly = true)

    public ActivityLogDTO getById(String id) {
        return activityLogRepository.findById(id).map(activityLogMapper::toDTO).orElseThrow(() -> new RuntimeException("ActivityLog not found: " + id));
    }




    private String resolveActor() {
        try {
            var auth = org.springframework.security.core.context.SecurityContextHolder.getContext().getAuthentication();
            if (auth != null && auth.isAuthenticated() && !"anonymousUser".equals(auth.getName())) {
                return auth.getName();
            }
        } catch (Exception e) {
            // Fall through to system default
        }
        return "system";
    }

}

File: yaml-to-code-generator/test/fixtures/vue-golden/src/domain/activityLog/activityLog.service.ts


import type { ActivityLog } from './activityLog.types';



import type { ActivityLogRepository } from './activityLog.repository';

export interface ActivityLogServiceHooks {



  beforeGetById?(id: string): Promise<void> | void;

  afterGetById?(result: ActivityLog): Promise<void> | void;


}

export class ActivityLogService {
  repository: ActivityLogRepository;
  private readonly hooks?: Partial<ActivityLogServiceHooks>;

  constructor(repository: ActivityLogRepository, hooks?: Partial<ActivityLogServiceHooks>) {
    this.repository = repository;
    this.hooks = hooks;
  }







  async getById(id: string): Promise<ActivityLog> {
    try { await this.hooks?.beforeGetById?.(id); } catch {}

    const result = await this.repository.getById(id);
    try { await this.hooks?.afterGetById?.(result); } catch {}
    return result;

  }


}
